src/plugins/data/common/search/expressions/esdsl.ts

```diff
@@ -0,0 +1,197 @@
+  typeof name,
+  Input,
+  Arguments,
+  Output
+>;
+
+/** @internal */
+export interface EsdslStartDependencies {
+  search: ISearchGeneric;
+  uiSettingsClient: UiSettingsCommon;
+}
+
+export const getEsdslFn = ({
+  getStartDependencies,
+}: {
+  getStartDependencies: (getKibanaRequest: any) => Promise<EsdslStartDependencies>;
+}) => {
+  const esdsl: EsdslExpressionFunctionDefinition = {
+    name,
+    type: 'es_raw_response',
+    inputTypes: ['kibana_context', 'null'],
+    help: i18n.translate('data.search.esdsl.help', {
+      defaultMessage: 'Run Elasticsearch request',
+    }),
+    args: {
+      dsl: {
+        types: ['string'],
+        aliases: ['_', 'q', 'query'],
+        help: i18n.translate('data.search.esdsl.q.help', {
+          defaultMessage: 'Query DSL',
+        }),
+        required: true,
+      },
+      index: {
+        types: ['string'],
+        help: i18n.translate('data.search.esdsl.index.help', {
+          defaultMessage: 'ElasticSearch index to query',
+        }),
+        required: true,
+      },
+      size: {
+        types: ['number'],
+        help: i18n.translate('data.search.esdsl.size.help', {
+          defaultMessage: 'ElasticSearch searchAPI size parameter',
+        }),
+        default: 10,
+      },
+    },
+    async fn(input, args, { inspectorAdapters, abortSignal, getKibanaRequest }) {
+      const { search, uiSettingsClient } = await getStartDependencies(getKibanaRequest);
+
+      const dsl = JSON.parse(args.dsl);
+
+      if (input) {
+        const esQueryConfigs = getEsQueryConfig(uiSettingsClient as any);
+        const query = buildEsQuery(
+          undefined, //        args.index,
+          input.query || [],
+          input.filters || [],
+          esQueryConfigs
+        );
+
+        if (!dsl.query) {
+          dsl.query = query;
+        } else {
+          query.bool.must.push(dsl.query);
+          dsl.query = query;
+        }
+      }
+
+      if (!inspectorAdapters.requests) {
+        inspectorAdapters.requests = new RequestAdapter();
+      }
+
+      const request = inspectorAdapters.requests.start(
+        i18n.translate('data.search.dataRequest.title', {
+          defaultMessage: 'Data',
+        }),
+        {
+          description: i18n.translate('data.search.es_search.dataRequest.description', {
+            defaultMessage:
+              'This request queries Elasticsearch to fetch the data for the visualization.',
+          }),
+        }
+      );
+
+      request.stats({
+        indexPattern: {
+          label: i18n.translate('data.search.es_search.indexPatternLabel', {
+            defaultMessage: 'Index pattern',
+          }),
+          value: args.index,
+          description: i18n.translate('data.search.es_search.indexPatternDescription', {
+            defaultMessage: 'The index pattern that connected to the Elasticsearch indices.',
+          }),
+        },
+      });
+
+      let res: IEsSearchResponse;
```

Comment:
It is not being used outside try block. I think it makes sense to declare that as a constant right in the try block.